### ISSUE TITLE
[Feature Request]: Discuss rolling out the <<resourceGroupName>> token across all pipeline parameter files

### ISSUE THREAD

**Issue by ahmadabdalla** [Azure/ResourceModules#1363]
### Description

We currently support a token called `<<resourceGroupName>` which is replaced by the `resourceGroupName` variable in the global variables YAML. 

If someone wants to change the resource group name to not be `validation-rg`, then they'd need to update all pipelines to do that as it is hardcoded in the parameter files. 

Should we roll this token out to all parameter files to allow changing the resource group name without impacting the parameter files?

**Comment by MariusStorhaug** [comment]
This is as requested by the issue: #953
Would be great to have _all_ combined into one, available as tokens or variables in the pipeline/workflows. So yes, you have my support! (smiley)

**Comment by ahmadabdalla** [comment]
@MariusStorhaug `global.variables.yml` will be `settings.yml` as per your suggestion.

**Comment by ahmadabdalla** [comment]
@MariusStorhaug on your point:
 _Ideally, it should be fully dynamic so that we have automation to just make all the variables in the global.variable.yml file available as tokens._

In the case of GitHub, we are taking the whole file and injecting it as environment variables. Is the above alluding to taking each environment variable, appending the token prefix and suffix and then searching for it in the file? like this:

<<envVariable1>>

**Comment by MariusStorhaug** [comment]
@ahmadabdalla, then I think I was unaware of how the tokens were made available in the piepline/workflows. My bad. You can disregard the point made on making them automatically available as env vars.

**Comment by MrMCake** [comment]
While I think this could be enabled at any time - I wonder if it is still relevant once the new dependency approach #1791 is merged for all modules. There, the resource group name is usually passed as a reference in to the test.

**Comment by ahmadabdalla** [comment]
@MrMCake with a recent feedback I got from a carml customer is that changing the RG name on the settings.yml doesn't reflect in the parameter files. Would there still be any use cases for the dependency pipelines to continue to be supported? If yes, then maybe we can roll this out?

**Comment by MrMCake** [comment]
> If yes, then maybe we can roll this out?
Currently, there would not be a use case for the dependency pipeline to be supported in the future.

**Comment by ahmadabdalla** [comment]
@MrMCake then we can close this ticket if there is no longer a need for it

**Comment by eriqua** [comment]
Moving back to needs triage to agree on this closure. I'd anyway suggest to open an issue to remove the resource group parameter from settings.yml and all related utility referencies.

**Comment by rahalan** [comment]
@eriqua please create the new issue as proposed and close the issue

**Comment by eriqua** [comment]
Created issue #2204 as agreed. Closing as not planned.
